opentelemetry-kotlin-implementation/src/commonMain/kotlin/io/embrace/opentelemetry/kotlin/provider/ApiProviderImpl.kt

```diff
@@ -19,27 +21,26 @@ import io.embrace.opentelemetry.kotlin.sync
 @OptIn(ExperimentalApi::class)
 @ThreadSafe
 internal class ApiProviderImpl<T>(
-    val supplier: (key: ApiProviderKey) -> T
+    val supplier: (key: InstrumentationScopeInfo) -> T
 ) {
 
-    private val map = mutableMapOf<ApiProviderKey, T>()
+    private val map = mutableMapOf<InstrumentationScopeInfo, T>()
 
-    fun getOrCreate(key: ApiProviderKey): T = sync(map) {
+    fun getOrCreate(key: InstrumentationScopeInfo): T = sync(map) {
         map.getOrPut(key) {
             supplier(key)
         }
     }
 
-    fun createKey(
+    fun createInstrumentationScopeInfo(
         attributes: MutableAttributeContainer.() -> Unit,
```

Comment:
nit: I'd move this parameter to the end, so it's the same order as the InstrumentationScopeInfo class (name, version, schemaUrl, attributes)